### PR TITLE
Allow overriding the span ref validation

### PR DESCRIPTION
**Pull request by randomswdev**
This pull request is a candidate for solving issue https://github.com/elastic/apm-agent-go/issues/843

### CODE REVIEW

File: module/apmot/tracer.go [hunk 1]
```diff
@@ -34,8 +34,14 @@ import (
 //
 // By default, the returned tracer will use apm.DefaultTracer.
 // This can be overridden by using a WithTracer option.
+// The option WithSpanRefValidator allows to override the
+// set of spans that are traced. By default only child-of
+// spans are traced.
```
Comment (axw) [inline module/apmot/tracer.go:39]:
```suggestion
// The option WithSpanRefValidator allows one to override the
// set of spans that are recorded. By default only child-of
// spans are recorded.
```

File: module/apmot/tracer.go [hunk 2]
```diff
@@ -197,6 +204,20 @@ func WithTracer(t *apm.Tracer) Option {
 	}
 }
 
+// SpanRefValidator verifies if a span is valid and can be processed.
```
Comment (axw) [inline module/apmot/tracer.go:207]:
```suggestion
// SpanRefValidator verifies if a span is valid and should be recorded.
```

**Review by axw — APPROVED** [review]
Thank you, this looks great! Just a couple of minor requested doc comment changes.

**Comment by axw** [comment]
jenkins run the tests please

File: module/apmot/tracer.go (hunk 2 above)
Comment (randomswdev) [reply module/apmot/tracer.go]: Changed in my last commit

File: module/apmot/tracer.go (hunk 1 above)
Comment (randomswdev) [reply module/apmot/tracer.go]: Changed in my last commit

**Comment by axw** [comment]
Thanks for your contribution @randomswdev, LGTM!

**Pull request merged**